### PR TITLE
Fileinputstream with treeset

### PR DESCRIPTION
**Pull request by EliosPride**

### CODE REVIEW

File: src/main/java/core/basesyntax/TreeSetCharacters.java [hunk 1]
```diff
@@ -19,7 +23,33 @@
...
+        }
+        try (FileInputStream fileInputStream = new FileInputStream(file)) {
+            while (fileInputStream.available() > 0) {
+                char alphabetChar = (char) fileInputStream.read();
+                if (Character.isLetter(alphabetChar)) {
+                    charSet.add(Character.toLowerCase(alphabetChar));
+                }
+            }
+            for (Character nextCharacter : charSet) {
+                charBuilder.append(nextCharacter);
```
Comment (al-pavlenko) [inline src/main/java/core/basesyntax/TreeSetCharacters.java:43]: stop append after 5 chars is appended. As result you will not need logic below of substing. Also you will be able to remove resultString var.

File: src/main/java/core/basesyntax/TreeSetCharacters.java [hunk 2]
```diff
@@ -19,7 +23,33 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws FileNotFoundException {
```
Comment (al-pavlenko) [inline src/main/java/core/basesyntax/TreeSetCharacters.java]: throws IOException

File: src/main/java/core/basesyntax/TreeSetCharacters.java [hunk 3]
```diff
@@ -19,7 +23,33 @@
...
+            }
+            resultString = charBuilder.toString();
+            if (charBuilder.length() > 5) {
+                return resultString.substring(0, 5);
+            }
+        } catch (FileNotFoundException e) {
+            System.out.println("File doesn't exist.");
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```
Comment (al-pavlenko) [inline src/main/java/core/basesyntax/TreeSetCharacters.java]: remove catch blocks. You need to throw exceptions if something went wrong.

File: src/main/java/core/basesyntax/TreeSetCharacters.java (hunk 2 above)
Comment (EliosPride) [reply src/main/java/core/basesyntax/TreeSetCharacters.java]: @al-pavlenko done.

File: src/main/java/core/basesyntax/TreeSetCharacters.java (hunk 3 above)
Comment (EliosPride) [reply src/main/java/core/basesyntax/TreeSetCharacters.java]: @al-pavlenko done.

File: src/main/java/core/basesyntax/TreeSetCharacters.java (hunk 1 above)
Comment (EliosPride) [reply src/main/java/core/basesyntax/TreeSetCharacters.java:43]: done

File: src/main/java/core/basesyntax/TreeSetCharacters.java [hunk 4]
```diff
@@ -19,7 +23,32 @@
  * Результат 2: acf</p>
  */
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+    public String getUniqueCharacters(String fileName) throws IOException {
+        TreeSet<Character> charSet = new TreeSet<>();
+        StringBuilder charBuilder = new StringBuilder();
+        int iterator = 5;
```
Comment (al-pavlenko) [inline src/main/java/core/basesyntax/TreeSetCharacters.java]: move to constant MAX_LENGTH

File: src/main/java/core/basesyntax/TreeSetCharacters.java [hunk 5]
```diff
@@ -19,7 +23,32 @@
...
+            while (fileInputStream.available() > 0) {
+                char alphabetChar = (char) fileInputStream.read();
+                if (Character.isLetter(alphabetChar)) {
+                    charSet.add(Character.toLowerCase(alphabetChar));
+                }
+            }
+            for (Character nextCharacter : charSet) {
+                charBuilder.append(nextCharacter);
+                iterator++;
+                if (iterator == 5) {
```
Comment (al-pavlenko) [inline src/main/java/core/basesyntax/TreeSetCharacters.java]: better to do on start of loop smth like that:
`if (charBuilder.length() == MAX_LENGTH) {
break;
}`
And return charBuilder.toString();
As result you don;t need to do extra operations of substrings by condition

Comment (EliosPride) [reply src/main/java/core/basesyntax/TreeSetCharacters.java]: @al-pavlenko don

File: src/main/java/core/basesyntax/TreeSetCharacters.java (hunk 4 above)
Comment (EliosPride) [reply src/main/java/core/basesyntax/TreeSetCharacters.java]: @al-pavlenko don

**Review by al-pavlenko — APPROVED** [review]

**Pull request merged**